lib/ramble/ramble/workspace/workspace.py

```diff
@@ -1148,6 +1148,62 @@ def create_mirror(self, mirror_root):
         self.input_mirror_cache = ramble.caches.MirrorCache(self.input_mirror_path)
         self.software_mirror_cache = ramble.caches.MirrorCache(self.software_mirror_path)
 
+    def simplify(self):
+        # First drop unused experiment templates from app dict so environments aren't rendered
+        app_dict = ramble.config.config.get_config('applications',
+                                                   scope=self.ws_file_config_scope_name())
+
+        # Build experiment sets to determine which templates never get used
+        self.software_environments = \
+            ramble.software_environments.SoftwareEnvironments(self)
+        experiment_set = self.build_experiment_set()
+
+        # This uses experiments dict directly because all_experiments() filters out templates
+        for exp_namespace in experiment_set.experiments.keys():
+            app_inst = experiment_set.get_experiment(exp_namespace)
+            if app_inst.is_template and not app_inst.generated_experiments:
+                app, wl, exp = exp_namespace.split('.')
```

Comment:
Here, it might be better to do:
```
app = app_inst.expander.application_name
wl = app_inst.expander.workload_name
exp = app_inst.expander.experiment_name
```

Just in case the formatting of the namespaces changes in the future.

Reply:
Done